packages/server/api/src/app/core/security/authorization-middleware.ts

```diff
@@ -0,0 +1,22 @@
+import { FastifyRequest } from 'fastify'
+import { principalTypeAuthz } from './authz/principal-type-authz'
+import { AuthorizationType, RouteKind } from '@activepieces/server-shared'
+import { securityUtils } from './utils'
+import { PrincipalType } from '@activepieces/shared'
+
+export const authorizationMiddleware = async (request: FastifyRequest): Promise<void> => {
+    const security = request.routeOptions.config?.security
+    
+    if (security.kind === RouteKind.PUBLIC) {
+        return
+    }
+
+    await principalTypeAuthz.authorize(request, security)
+    
+    if (security.authorization.type === AuthorizationType.PROJECT && (request.principal.type === PrincipalType.USER || request.principal.type === PrincipalType.SERVICE)) {
```

Comment:
Why do we have to check for request.princpal.type , isn't it already enforced at the type level when authorizationType is project?